src/main.rs

```diff
@@ -144,10 +144,16 @@ fn write_to_file(file: impl AsRef<Path>) {
             return;
         }
     }
-    generate_cargo_nix(fs::File::create(&file).expect(&format!(
-        "could not open file for writing: {}",
-        file.display()
-    )));
+
+    let mut temp_file = tempfile::Builder::new()
+        .tempfile()
+        .expect("could not create new temporary file");
+
+    generate_cargo_nix(&mut temp_file);
+
+    temp_file
+        .persist(path)
+        .unwrap_or_else(|_| panic!("could not write file to final location: {}", path.display()));
```

Comment:
Regarding the `_` thingy: maybe we should print the error instead of ignoring it?